pkg/operations/switchover_util.go

```diff
@@ -57,14 +60,15 @@ func needDoSwitchover(ctx context.Context,
 	case KBSwitchoverCandidateInstanceForAnyPod:
 		return true, nil
 	default:
-		pods, err := component.ListOwnedPods(ctx, cli, synthesizedComp.Namespace, synthesizedComp.ClusterName, synthesizedComp.Name)
-		if err != nil {
+		targetPod := &corev1.Pod{}
+		if err := cli.Get(ctx, client.ObjectKey{Name: switchover.InstanceName, Namespace: synthesizedComp.Namespace}, targetPod); err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, controllerutil.NewFatalError(err.Error())
+			}
 			return false, err
 		}
-		podParent, _ := instanceset.ParseParentNameAndOrdinal(pod.Name)
-		siParent, o := instanceset.ParseParentNameAndOrdinal(switchover.InstanceName)
-		if podParent != siParent || o < 0 || o >= len(pods) {
-			return false, errors.New("switchover.InstanceName is invalid")
+		if targetPod.Labels[constant.AppInstanceLabelKey] != synthesizedComp.ClusterName || component.GetComponentNameFromObj(targetPod) != switchover.ComponentName {
```

Comment:
check targetPod.labels not nil

Reply:
if no set operation, golang can ignore the map is nil and return an empty value